backend/core/src/listings/dto/listing.dto.ts

```diff
@@ -70,6 +71,22 @@ export class ListingDto extends OmitType(Listing, [
   @Type(() => ApplicationMethodDto)
   applicationMethods: ApplicationMethodDto[]
 
+  @Expose()
+  @IsOptional({ groups: [ValidationsGroupsEnum.default] })
+  @Transform(
```

Comment:
1. That could be handled with a getter instead of transform which is more readable (I think transforms are more useful when we would like to create a new property out of input, but here referallApplication is just a derivative of value already assigned to ListingDto anyway)
2. Retrieval logic could be simplified by using [Array.find](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/find) directly

Reply:
Updated to use getter and Array.find